pkg/route/ingress/ingress.go

```diff
@@ -973,3 +1004,22 @@ func destinationCACertificateForIngress(ingress *networkingv1.Ingress, secretLis
 	}
 	return nil
 }
+
+// shouldPropagateLabelsToRoute verifies if annotation map contains the key
+// 'router.openshift.io/reconcile-labels' and if its value is a non-empty
+// parseable boolean (true, false, t, T, f, F, etc).
+// In case the annotation exists but the value is invalid, it will return "false"
+// and the parsing error.
+// Otherwise it returns the underlying boolean value.
+func shouldPropagateLabelsToRoute(annotations map[string]string) (bool, error) {
+	propagateLabelsFlag, ok := annotations[routecontroller.PropagateIngressLabelFlag]
+	if !ok {
+		return false, nil
+	}
+
+	shouldPropagate, err := strconv.ParseBool(strings.ToLower(propagateLabelsFlag))
```

Comment:
```suggestion
	shouldPropagate, err := strconv.ParseBool(strings.TrimSpace(propagateLabelsFlag))
```
`strconv.ParseBool ` supports mixed case, if we would like to be safe we can trim whitespaces to avoid errors in case of ' true '

Reply:
good call, let me do it